pulser-core/pulser/sequence/sequence.py

```diff
@@ -681,10 +692,26 @@ def check_retarget(ch_obj: Channel) -> bool:
             for call in self._calls + self._to_build_calls
             if call.name == "enable_eom_mode"
         ]
-        for old_ch_name, old_ch_obj in self.declared_channels.items():
+        all_channels_new_device = {
+            **new_device.channels,
+            **new_device.dmm_channels,
+        }
+        all_declared_channels = self.declared_channels
+        if self.is_parametrized():
+            # SLM mask configures a DMM after all the DMMs
+            for call in self._to_build_calls:
+                if call.name == "config_slm_mask":
```

Comment:
In a `strict` switch, can't the `bottom_detuning` influence the value of the DMM pulse for the SLM mask? Shouldn't we check for that?

Reply:
Ah yes I forgot about this case.

Reply:
I also want to add a test for the Mappable Register !
I will first solve the PR on the "Fix Error Messages for switch device" separately.
I will also change the name of `modulate_det_map` to` add_to_dmm`in `add_dmm_sequence`